private_set_intersection/python/psi_bindings.cpp

```diff
@@ -22,27 +22,73 @@ T throwOrReturn(const private_join_and_compute::StatusOr<T>& in) {
   return in.ValueOrDie();
 }
 
+template <class T>
+auto saveProto(const T& obj) {
+  char* output = (char*)malloc(obj.ByteSizeLong() * sizeof(char));
```

Comment:
This seems a bit complicated. You can use a C++ `std::string` as a buffer instead of C-style `malloc`, and pass it directly to the `py::bytes` constructor:
```C++
auto saveProto(const T& obj) {
  return py::bytes(obj.SerializeAsString());
}
```